__tests__/files/file.spec.ts

```diff
@@ -141,3 +157,57 @@ describe('File data change', () => {
 		expect(file.root).toBe('/files/emma')
 	})
 })
+
+
+describe('Altering attributes updates mtime', () => {
+	test('mtime is updated on existing attribute', () => {
+		const file = new File({
+			source: 'https://cloud.domain.com/remote.php/dav/files/emma',
+			mime: 'image/jpeg',
+			owner: 'emma',
+			mtime: new Date(Date.UTC(1990, 0, 1, 0, 0, 0)),
+			attributes: {
+				test: true
+			},
+		})
+		expect(file.attributes.test).toBe(true)
+		file.attributes.test = false
+	
+		// Check that mtime has been updated
+		expect(file.mtime?.getDate()).toBe(new Date().getDate())
+		expect(file.attributes.test).toBe(false)
+	})
+
+	test('mtime is updated on new attribute', () => {
+		const file = new File({
+			source: 'https://cloud.domain.com/remote.php/dav/files/emma',
+			mime: 'image/jpeg',
+			owner: 'emma',
+			mtime: new Date(Date.UTC(1990, 0, 1, 0, 0, 0)),
+		})
+		expect(file.attributes.test).toBeFalsy()
+		file.attributes.test = true
+	
+		// Check that mtime has been updated
+		expect(file.mtime?.getDate()).toBe(new Date().getDate())
```

Comment:
this test will fail occasionally if the previous statement runs a microsecond earlier than the next invocation of `new Date`

Reply:
No because I'm comparing the date, so it would have to be exactly 1ms before midnight :grin: